gettsim/interface.py

```diff
@@ -531,6 +538,78 @@ def _root_nodes(dag):
             yield node
 
 
+def _add_rounding_for_one_function(base, direction):
+    """Decorator to round the output of a function.
+
+    Parameters
+    ----------
+    base : float
+        Precision of rounding (e.g. 0.1 to round to the first decimal place)
+    round_d : bool
+        Whether rounding should be applied
+    direction : str
+        Whether the series should be rounded up, down or to the nearest number
+
+    Returns
+    -------
+    results : pandas.Series
+        Series with (potentially) rounded numbers
+    """
+
+    def inner(func):
+        # Make sure that signature is preserved
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            out = func(*args, **kwargs)
+            if direction == "up":
+                return base * np.ceil(out / base)
+            elif direction == "down":
+                return base * np.floor(out / base)
+            elif direction == "nearest":
+                return base * (out / base).round()
```

Comment:
I prefer if each function has just one return statement in the last row. I.e. just assign a variable inside the if conditions and return it once at the end.